components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -0,0 +1,189 @@
+        }
+    }
+
+    handleA11yDeactivateEvent = () => {
+        if (this.categoryTitleRef.current) {
+            this.categoryTitleRef.current.removeEventListener('keydown', this.handleA11yKeyDown);
+        }
+    }
+
+    handleA11yKeyDown = (e: KeyboardEvent) => {
+        if (isKeyPressed(e, Constants.KeyCodes.ENTER)) {
+            this.handleCollapse();
+        }
+    }
+
+    renderChannel = (channel: Channel) => {
+        const {isCollapsed} = this.props;
+
+        return (
+            <SidebarChannel
+                channelId={channel.id}
+                setChannelRef={this.props.setChannelRef}
+                getChannelRef={this.props.getChannelRef}
+                isCategoryCollapsed={isCollapsed}
+            />
+        );
+    }
+
+    handleCollapse = () => {
+        const {category, isCollapsed} = this.props;
+
+        if (isCollapsed) {
+            trackEvent('ui', 'ui_sidebar_expand_category');
+        } else {
+            trackEvent('ui', 'ui_sidebar_collapse_category');
+        }
+
+        this.props.actions.setCategoryCollapsed(category.id, !isCollapsed);
+    }
+
+    handleOpenDirectMessagesModal = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.stopPropagation();
+        this.props.handleOpenMoreDirectChannelsModal(e.nativeEvent);
+    }
+
+    render() {
+        const {category, isCollapsed, channels} = this.props;
+        if (!category) {
+            return null;
+        }
+
+        if (category.type !== CategoryTypes.DIRECT_MESSAGES && (!channels || !channels.length)) {
+            return null;
+        }
+
+        const renderedChannels = channels.map(this.renderChannel);
+
+        let directMessagesModalButton;
+        let hideArrow = false;
+        if (category.type === CategoryTypes.DIRECT_MESSAGES) {
+            const helpLabel = localizeMessage('sidebar.createDirectMessage', 'Create new direct message');
+
+            const tooltip = (
+                <Tooltip
+                    id='new-group-tooltip'
+                    className='hidden-xs'
+                >
+                    {helpLabel}
+                </Tooltip>
+            );
+
+            directMessagesModalButton = (
+                <button
+                    className='SidebarChannelGroupHeader_addButton'
+                    onClick={this.handleOpenDirectMessagesModal}
+                    aria-label={helpLabel}
+                >
+                    <OverlayTrigger
+                        delayShow={500}
+                        placement='top'
+                        overlay={tooltip}
+                    >
+                        <i className='icon-plus'/>
+                    </OverlayTrigger>
+                </button>
+            );
+
+            if (!channels || !channels.length) {
+                hideArrow = true;
+            }
+        }
+
+        let displayName = category.display_name;
```

Comment:
Don't we wanna localize display names if they are not custom?

Reply:
See the line below, we are doing that.

Reply:
My bad. I read it wrong 😀